tests/autoblocks/test_tracer.py

```diff
@@ -448,3 +441,215 @@ def test_tracer_start_span(*args, **kwargs):
 
     assert tracer._properties.get("span_id") is None
     assert tracer._properties.get("parent_span_id") is None
+
+
+@mock.patch.object(
+    uuid,
+    "uuid4",
+    side_effect=["mock-uuid-1"],
+)
+def test_tracer_prod_evaluations(httpx_mock):
+    class MyEvaluator(BaseEventEvaluator):
+        id = "my-evaluator"
+
+        def evaluate_event(self, event: TracerEvent) -> Evaluation:
+            return Evaluation(
+                score=0.9,
+                threshold=Threshold(gte=0.5),
+            )
+
+    httpx_mock.add_response(
+        url=INGESTION_ENDPOINT,
+        method="POST",
+        status_code=200,
+        json={"traceId": "my-trace-id"},
+        match_headers={"Authorization": "Bearer mock-ingestion-key"},
+        match_content=make_expected_body(
+            dict(
+                message="my-message",
+                traceId="my-trace-id",
+                timestamp=timestamp,
+                properties={
+                    "evaluations": [
+                        {
+                            "id": "mock-uuid-1",
+                            "score": 0.9,
+                            "metadata": None,
+                            "threshold": {"lt": None, "lte": None, "gt": None, "gte": 0.5},
+                            "evaluatorExternalId": "my-evaluator",
+                        }
+                    ]
+                },
+            )
+        ),
+    )
+    tracer = AutoblocksTracer("mock-ingestion-key")
+    tracer.send_event(
+        "my-message",
+        trace_id="my-trace-id",
+        timestamp=timestamp,
+        properties={},
+        evaluators=[MyEvaluator()],
+    )
+
+
+@mock.patch.object(
+    uuid,
+    "uuid4",
+    side_effect=["mock-uuid" for _ in range(2)],
+)
+def test_tracer_prod_async_evaluations(httpx_mock):
+    class MyEvaluator1(BaseEventEvaluator):
+        id = "my-evaluator-1"
+
+        async def evaluate_event(self, event: TracerEvent):
```

Comment:
should we add some `await asyncio.sleep`s here to catch the issues you were seeing with `return await`?

Reply:
oh right, gotcha